src/OidcKeyRegistry.sol

```diff
@@ -79,4 +68,55 @@ contract OidcKeyRegistry is Initializable, OwnableUpgradeable {
   function getKeys(bytes32 issHash) public view returns (Key[MAX_KEYS] memory) {
     return OIDCKeys[issHash];
   }
+
+  function deleteKey(bytes32 issHash, bytes32 kid) public onlyOwner {
+    _deleteKey(issHash, kid);
+    _compactKeys(issHash);
+    emit KeyDeleted(issHash, kid);
+  }
+
+  function _compactKeys(bytes32 issHash) private {
+    Key[MAX_KEYS] memory keys;
+    uint8 keyCount = 0;
+    uint8 currentIndex = keyIndexes[issHash];
+
+    for (uint8 i = 0; i < MAX_KEYS; i++) {
+      uint8 circularIndex = (currentIndex + i) % MAX_KEYS;
+      if (OIDCKeys[issHash][circularIndex].kid != 0) {
+        keys[keyCount] = OIDCKeys[issHash][circularIndex];
+        keyCount++;
+      }
+    }
+
+    for (uint8 i = 0; i < keyCount; i++) {
+      OIDCKeys[issHash][i] = keys[i];
+    }
+
+    for (uint8 i = keyCount; i < MAX_KEYS; i++) {
+      delete OIDCKeys[issHash][i];
+    }
```

Comment:
I feel that the comment at the beggining of each block explaining what it do might be appreciated by the next person reading this. The indexes starting and ending at key count are key and easy to skip in a quick look.